code/src/components/TodoInput.js

```diff
@@ -0,0 +1,62 @@
+import React, { useState } from "react"
+import { useDispatch } from "react-redux"
+import { todos } from "../reducers/todos"
+import styled from 'styled-components'
+import { Button } from '../lib/Button'
+
+
+export const TodoInput = ({ listId }) => {
+    const dispatch = useDispatch();
+    const [inputValue, setInputValue] = useState("");
+
+    const handleOnSumbit = e => {
+        e.preventDefault();
+ 
+
+    dispatch (
+        todos.actions.addTodo({
+            listId: listId,
+            itemInfo: { description: inputValue, done: false }
+        })
```

Comment:
The indentation here makes me thing that the dispatch happens outside of the `handleOnSubmit`